File: aoc2015/src/test/java/hu/wolfman/aoc2015/Day03Test.java
package hu.wolfman.aoc2015;

import org.junit.jupiter.api.Test;

import java.util.Collections;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class Day03Test {

    @Test
    public void testPartOne() {
        Day day = new Day03();
        assertEquals(2, day.solvePartOne(Collections.singletonList(">")));
        assertEquals(4, day.solvePartOne(Collections.singletonList("^>v<")));
        assertEquals(2, day.solvePartOne(Collections.singletonList("^v^v^v^v^v")));
    }

    @Test
    public void testPartTwo() {
        Day day = new Day03();
        assertEquals(3, day.solvePartTwo(Collections.singletonList("^v")));
        assertEquals(3, day.solvePartTwo(Collections.singletonList("^>v<")));
        assertEquals(11, day.solvePartTwo(Collections.singletonList("^v^v^v^v^v")));
    }


}

File: aoc2021/src/test/kotlin/Day03Test.kt
import org.junit.Test
import kotlin.test.assertEquals

class Day03Test {
    @Test
    fun testPartOne() {
        val list = readInput("day03test")
        assertEquals(198, Day03.solvePartOne(list))
    }

    @Test
    fun testPartTwo() {
        val list = readInput("day03test")
        assertEquals(230, Day03.solvePartTwo(list))
    }
}